packages/SwingSet/src/kernel/kernel.js

```diff
@@ -577,10 +578,12 @@ export default function buildKernel(kernelEndowments) {
    * error message for the problem.
    */
   function createVatDynamically(buildFnSrc) {
-    const endowments = { require: kernelRequire };
-    const buildFn = evaluateProgram(buildFnSrc, endowments);
+    const endowments = { require: kernelRequire, HandledPromise };
+    const nameSpace = evaluateProgram(buildFnSrc.source, endowments)();
+    const buildFn = () => harden(nameSpace);
```

Comment:
I'm a bit confused about this zero-argument `buildFn`. How do the `E` (and sometimes `D`) functions get into the new vat? I'm guessing that `evaluateProgram` is responsible?

Reply:
(As discussed, and for posterity)

Vats now support only `~.`, so there's no need to supply E. Dynamic vats shouldn't need access to D.

The way that bundle-source and the contractHost process the vat code turns `~.` into reliance on HandledPromise, which is made available to the evaluated code.

Reply:
> Vats now support only ~., so there's no need to supply E

Actually there is: `E.unwrap`.

Reply:
Given mfig's comment that E can be imported, it doesn't appear that it's necessary to provide E any longer.